qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SnapshotBasedRecoveryIT.java

```diff
@@ -65,16 +73,38 @@ public void testSnapshotBasedRecovery() throws Exception {
                 break;
             case MIXED:
             case UPGRADED:
-                // the following `if` for first round mixed was added as a selective test mute. Sometimes the primary shard ends
-                // on the upgraded node. This causes issues when removing and adding replicas, since then we cannot allocate to
-                // any of the old nodes. That is an issue only for the first mixed round, hence this check.
-                // Ideally we would find the reason the primary ends on the upgraded node and fix that (or figure out that it
-                // is all good).
-                // @AwaitsFix(bugUrl = https://github.com/elastic/elasticsearch/issues/76595)
-                if (CLUSTER_TYPE != MIXED || FIRST_MIXED_ROUND == false) {
-                    // Drop replicas
-                    updateIndexSettings(indexName, Settings.builder().put(IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 0));
+                if (FIRST_MIXED_ROUND) {
+                    String primaryNodeId = getPrimaryNodeIdOfShard(indexName, 0);
```

Comment:
I wonder if we should not instead just find the node on the new version and put the exclusion in always? I think there is a risk of the shard moving between the check here and adding replicas below.